PeerioTechnologies/peerio-desktop
# feat: UI feedback for poor connection when downloading inline image preview

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### PR author

#### Relevant issue tracker links  
https://peerio.tpondemand.com/entity/7652
#### Linked PRs in other repos
PR for similar functionality for URL previews:
https://github.com/PeerioTechnologies/peerio-desktop/pull/160
#### Manual testing instructions  
"Poor connection..." message should appear when client app is trying to download an inline image preview, but app is offline. It should not appear if: image is not supposed to download (e.g. oversize, or user has disabled previews), or if user is online.
#### Additional notes
The mockups show the progress bar tracking with how much of the file has been downloaded. My experience (on pretty fast internet) is that the download is so quick that you almost never see the download bar progress, just a very quick flash right before the download completes and the image shows up. So maybe it would be better to switch this to the "indefinite" style progress bar, as is used for URL previews.

----
### Reviewer

* Localisation.
* Architectural decisions.
* Performance.
* Comments.
* Big feature tech design documentation.
* Readability.
* DRY.
* Can we extract something to SDK?
* Are there any bicycles invented that should be replaced with existing SDK
* Are there unit and e2e tests added?

----
### Repository owner
